hypervisor/src/mshv/mod.rs

```diff
@@ -905,17 +906,27 @@ impl vm::Vm for MshvVm {
     /// Start logging dirty pages
     ///
     fn start_dirty_log(&self) -> vm::Result<()> {
-        Err(vm::HypervisorVmError::StartDirtyLog(anyhow!(
-            "functionality not implemented"
-        )))
+        self.fd
+            .enable_dirty_page_tracking()
+            .map_err(|e| vm::HypervisorVmError::EnableDirtyPageTracking(e.into()))
     }
     ///
     /// Stop logging dirty pages
     ///
     fn stop_dirty_log(&self) -> vm::Result<()> {
-        Err(vm::HypervisorVmError::StopDirtyLog(anyhow!(
-            "functionality not implemented"
-        )))
+        let dirty_log_slots = self.dirty_log_slots.read().unwrap();
+        // Before disabling the dirty page tracking we need
+        // to set the dirty bits in the Hypervisor
+        // This is a requirement from Microsoft Hypervisor
```

Comment:
Question: what about the "read-only" regions?

Reply:
No matter read-only or not we keep tracking of the slots. For read-only hypervisor can enable the dirty bits,
        but a VM exit happens before setting the dirty bits. I will change the commit.